FrontEnd/src/components/ProfilePage/FormComponents/GeneralInfo.js

```diff
@@ -165,6 +176,80 @@ const GeneralInfo = (props) => {
         }
     };
 
+    const uploadImage = async (url, imageKey, image) => {
+        if (image instanceof File || image === '') {
+            const formData = new FormData();
+            formData.append(imageKey, image);
+            const token = localStorage.getItem('Token');
+            try{
+                const response = await fetch(url, {
+                    method: 'PUT',
+                    headers: {
+                        'Authorization': `Token ${token}`,
+                    },
+                    body: formData,
+                });
+                if (response.status === 200) {
+                    const data = await response.json();
+                    profileMutate((prevState) => {
+                        return { ...prevState, [imageKey]: data[imageKey] };
+                    });
+                    if (imageKey === 'banner_image') {
+                        setBannerImage(mainProfile.banner_image);
+                    } else {
+                        setLogoImage(mainProfile.logo_image);
+                    }
+                    data[imageKey] === null
+                        ? toast.success(imageKey === 'banner_image' ? 'Банер видалено з профілю' : 'Логотип видалено з профілю')
+                        : toast.success(imageKey === 'banner_image' ? 'Банер успішно додано у профіль' : 'Логотип успішно додано у профіль');
+                } else if (response.status === 400) {
+                    toast.error('Не вдалося завантажити банер/лого, сталася помилка');
+                }
+            } catch (error) {
+                console.error('Error uploading image:', error);
+            }
+        }
+    };
+
+    const onUpdateImageField = (e) => {
+        const file = e.target.files[0];
+        e.target.value = '';
+        const imageUrl = URL.createObjectURL(file);
+        if (file) {
+            if (e.target.name === 'banner_image') {
+                if (file.size > BANNER_IMAGE_SIZE) {
+                    setBannerImageError('Максимальний розмір файлу 5 Mb');
+                } else {
+                    setBannerImageError(null);
+                    setBannerImage(file);
+                    setProfile((prevState) => {
+                        const newState = { ...prevState, [e.target.name]: imageUrl };
+                        return newState;
+                    });
+                }
+            } else {
+                if (file.size > LOGO_IMAGE_SIZE) {
+                    setLogoImageError('Максимальний розмір файлу 1 Mb');
+                } else {
+                    setLogoImageError(null);
+                    setLogoImage(file);
+                    setProfile((prevState) => {
+                        const newState = { ...prevState, [e.target.name]: imageUrl };
+                        return newState;
+                    });
+                }
+            }
+        }
```

Comment:
So many if/else clauses, could we try to break them into a few functions?

Reply:
Done